test/prog.c

```diff
@@ -401,4 +412,58 @@ prog_is_stopped (void)
     return prog_stopped != 0;
 }
 
-
+//Partly taken from https://www.binarytides.com/hostname-to-ip-address-c-sockets-linux/
+int getIpfromDNS(const char* hostname, char* ipaddr, bool version, const char* port)
+{
+	struct addrinfo hints, *servinfo, *p;
+	struct sockaddr_in *h;
+	struct sockaddr_in6 *h6;
+	int rv;
+	char * port2 = strdup(port); /*so that it's no longer const */
+	char * hostname2 = strdup(hostname);
+
+	memset(&hints, 0, sizeof(hints));
+	if (version)
+	{
+		hints.ai_family = AF_INET6;
+	}
+	else
+	{
+		hints.ai_family = AF_INET;
+	}
+	
+	hints.ai_socktype = SOCK_STREAM;
+	
+
+	if ((rv = getaddrinfo(hostname2, port2 , &hints, &servinfo)) != 0)
+	{
+		fprintf(stderr, "getaddrinfo: %s\n", gai_strerror(rv));
```

Comment:
The project-standard `LSQ_ERROR()` should be used instead of `fprintf(stderr, ...)`.